### PR TITLE
Add retree exercise

### PR DESCRIPTION
**Pull request by fidelcoria**
Implemented the Retree exercise in Prolog. I think this might be blocked by https://github.com/exercism/website-icons/issues/22

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -147,6 +147,18 @@
       "unlocked_by": null,
       "difficulty": 1,
       "topics": []
+    },
+    {
+      "slug": "retree",
+      "uuid": "27856806-0dee-409f-8047-357de07b914d",
+      "core": false,
+      "unlocked_by": null,
+      "difficulty": 3,
```
Comment (Average-user) [inline config.json]: Difficulty should be 1 since all other exercises are. But would be nice to have another PR to add proper difficulty level for all exercises.

**Comment by Average-user** [comment]
As a suggestion I think it would have been nice if the exercise also included post-order traversal.  That way you can avoid the multiple solutions issue. Since given two of the 3 traversals, there is only one way of constructing the tree if I'm not mistaken

**Comment by fidelcoria** [comment]
If the tree does contains repeated elements then multiple solutions are always possible. Including post-order traversals does not make a difference. Multiple solutions is not an "issue," especially for Prolog. On the contrary, it is an opportunity to practice with non-determinism.

**Comment by Average-user** [comment]
@fidelcoria thanks for the PR
